src/Roslyn.Diagnostics.Analyzers/Core/DeclarePublicAPIAnalyzer.Impl.cs

```diff
@@ -507,16 +507,34 @@ internal List<ApiLine> GetDeletedApiList()
 
             private bool IsPublicAPI(ISymbol symbol)
             {
+
                 if (symbol is IMethodSymbol methodSymbol && s_ignorableMethodKinds.Contains(methodSymbol.MethodKind))
                 {
                     return false;
                 }
-
                 // We don't consider properties to be public APIs. Instead, property getters and setters
                 // (which are IMethodSymbols) are considered as public APIs.
-                if (symbol is IPropertySymbol)
+                // Unless property is an auto-property.
+                if (symbol is IPropertySymbol propertySymbol)
                 {
-                    return false;
+                    if (symbol.Language != "Visual Basic") return false;
+                    // Write-Only auto-properties are not allowed. (as of VB 15.0)
+                    if (propertySymbol.IsWriteOnly)
+                    {
+                        //if (!propertySymbol.SetMethod.IsImplicitlyDeclared)
+                        //{
+                        //    return false;
+                        //}
+                        return false;
+                    };
+                    if (propertySymbol.IsReadOnly && !propertySymbol.GetMethod.IsImplicitlyDeclared)
+                    {
+                        return false;
+                    }
+                    if (!propertySymbol.GetMethod.IsImplicitlyDeclared && !propertySymbol.SetMethod.IsImplicitlyDeclared)
```

Comment:
>!propertySymbol.SetMethod.IsImplicitlyDeclared [](start = 74, length = 46)

Nit: could collapse this and the previous `if` to `(!propertySymbol.GetMethod.IsImplicitlyDeclared && (propertySymbol.IsReadOnly || !propertySymbol.SetMethod.IsImplicitlyDeclared))`